producer/event_producer.py

```diff
@@ -118,12 +147,98 @@ def msg_processor(self, msg):
         msg.cursor.send_feedback(flush_lsn=msg.data_start)
         self.check_shutdown()
 
+    @cachetools.cached(cache=table_name_cache, key=lambda self, relation_id: cachetools.keys.hashkey(relation_id))
+    def __get_table_name(self, relation_id: int) -> str:
+        """
+        Get the table name using the relation ID.
+
+        Args:
+            relation_id (int): The relation ID of the table.
+
+        Returns:
+            str: Full table name including schema.
+        """
+        try:
+            logger.debug('Getting table name for relation ID: %s', relation_id)
+            self.__db_cur.execute(
+                "select schemaname, relname from pg_stat_user_tables where relid = %s;",
+                (relation_id,)
+            )
+            schema_name, table_name = self.__db_cur.fetchone()
+            return f'{schema_name}.{table_name}'
+        except Exception:
+            logger.exception("Failed to get table name.")
+            raise
+    
+    @cachetools.cached(cache=schema_cache, key=lambda self, relation_id: cachetools.keys.hashkey(relation_id))
+    def __get_schema(self, relation_id) -> dict:
+        """
+        Retrieve the schema for the relation.
+        :return: A dictionary containing the schema information.
+        """
+        logger.debug(f'Relation ID: {relation_id}')
+        logger.debug('Getting Schema...')
+
+        schema = {
+            'relation_id': relation_id,
+            'columns': []
+        }
+
+        logger.debug('Getting column names and types...')
+        self.__db_cur.execute(
+            f'SELECT attname, atttypid FROM pg_attribute WHERE attrelid = {relation_id} AND attnum > 0;'
+        )
+
+        for column in self.__db_cur.fetchall():
+            schema['columns'].append({
+                'name': column[0],
+                'type': column[1]
+            })
+
+        logger.debug(f'Schema retrieved successfully. {schema}')
+        return schema
+
+    def pgoutput_msg_processor(self, msg):
+        message_type = msg.payload[:1].decode('utf-8')
+
+        # Convert bytes to int
+        relation_id = parser_utils.convert_bytes_to_int(msg.payload[1:5])
+
+        if message_type in ['I', 'U', 'D']:
+            operation_type = 'INSERT' if message_type == 'I' else 'UPDATE' if message_type == 'U' else 'DELETE'
+            table_name = self.__get_table_name(relation_id)
+
+            if table_name in self.__pg_tables:
+                logger.debug(f'{operation_type} Change occurred on table: {table_name}')
+                logger.debug(f'{operation_type} Change occurred at LSN: {msg.data_start}')
+
+                payload = {
+                    'schema': self.__get_schema(relation_id),
+                    'payload': base64.b64encode(msg.payload).decode('utf-8')
```

Comment:
encoding to base64 and decoding to convert to string
Bytes data was not being compressed